### PR TITLE
Document and streamline k8s.io site deploy

### PR DESCRIPTION
**Pull request by thockin**
1) Retool the canary.sh script into deploy.sh.
2) Use `rollout restart` instead of scale to 0.
3) Document how to do a rollout.

### CODE REVIEW

File: k8s.io/README.md [hunk 1]
```diff
@@ -57,23 +57,18 @@ Redirections
...
 ====
-Configure kubectl to target a test cluster on GKE.
 
-Run `make deploy-fake-secret deploy` and wait for the service to be available--
-the load balancer may take some time to configure.
+1) Log into Google Cloud Shell.  Our clusters do not allow access from the
+   internet.
 
-Set `TARGET_IP` to the ingress IP of the running service:
+2) Configure `kubectl` to have a cluster named "prod-aaa".  Running
```
Comment (dims) [inline k8s.io/README.md:66]: @thockin using the console the `connect` button tells me that i should use:
```gcloud container clusters get-credentials aaa --region us-central1 --project kubernetes-public```

When i do that, the name is `gke_kubernetes-public_us-central1_aaa`

So then i'd need to do `kubectl --cluster=gke_kubernetes-public_us-central1_aaa get ns` 

what's the best way to get the name to be what the script expects - namely `prod-aaa`?

File: k8s.io/deploy.sh [hunk 2]
```diff
@@ -18,8 +18,23 @@ set -o errexit
...
+fi
+if [ "$1" != "canary" -a "$1" != "prod" ]; then
+    echo "unsupported target '$1'"
+    echo "usage: $0 [ canary | prod ]"
+    exit 1
+fi
+
+NS="k8s-io-$1"
+
+echo "Deploying to namespace $NS - ^C now to abort..."
```
Comment (dims) [inline k8s.io/deploy.sh:33]: this is a handy tip. thanks!

**Comment by thockin** [comment]
Good point.  I did that manually.  Will make it easier.

**Comment by thockin** [comment]
take a look now.

File: k8s.io/README.md (hunk 1 above)
Comment (thockin) [reply k8s.io/README.md]: Good point.  I did that manually.  Will make it easier.  Take a look now

File: k8s.io/deploy.sh [hunk 3]
```diff
@@ -28,26 +44,32 @@ kc apply \
...
+kc rollout restart deployment k8s-io
 
 echo "waiting for all replicas to be up"
 while true; do
   sleep 3
-  read WANT HAVE < <( \
+  read WANT HAVE UNAVAIL < <( \
       kc get deployment k8s-io \
-          -o go-template='{{.spec.replicas}} {{.status.readyReplicas}}{{"\n"}}'
+          -o go-template='{{.spec.replicas}} {{.status.readyReplicas}} {{.status.unavailableReplicas}}{{"\n"}}'
```
Comment (dims) [inline k8s.io/deploy.sh:54]: tried this by hand, looks good

File: k8s.io/deploy.sh [hunk 4]
```diff
@@ -28,26 +44,32 @@ kc apply \
...
+  if [ -n "${WANT}" -a -n "${HAVE}" -a "${WANT}" == "${HAVE}" -a "${UNAVAIL}" == 0 ]; then
     break
   fi
-  echo "  want ${WANT}, found ${HAVE} ready"
+  echo "  want ${WANT}, found ${HAVE} ready and ${UNAVAIL} unavailable"
 done
 
-make test TARGET_IP=34.102.239.89
+# We can only test IPv4 from within GCP.
+all_ips=$(kc get ing k8s-io -o go-template='{{range .status.loadBalancer.ingress}}{{.ip}}{{end}}')
```
Comment (dims) [inline k8s.io/deploy.sh:70]: tested this by hand. looks good

File: k8s.io/deploy.sh [hunk 5]
```diff
@@ -18,8 +18,24 @@ set -o errexit
...
+    exit 1
+fi
+if [ "$1" != "canary" -a "$1" != "prod" ]; then
+    echo "unsupported target '$1'"
+    echo "usage: $0 [ canary | prod ]"
+    exit 1
+fi
+
+NS="k8s-io-$1"
+CONTEXT="gke_kubernetes-public_us-central1_aaa"
```
Comment (dims) [inline k8s.io/deploy.sh:32]: thanks for this default!

**Review by dims — APPROVED** [review]
LGTM

**Comment by dims** [comment]
/approve

@spiffxp can you please `/lgtm`

**Comment by dims** [comment]
whoops! :( i had LGTM first in code review and then did `/approve`, will be more careful!

**Pull request merged**